### ISSUE TITLE
Supermarket reconfigure fails in airgapped environment, ignores rubygems_url

### ISSUE THREAD

**Issue by flakzilla** [chef/supermarket#1712]
Attempted upgrade to supermarket 3.1.51 in airgapped environment because of #1705 using supermarket-omnibus-cookbook. Chef run to perform the upgrade failed during `supermarket-ctl reconfigure` step with error message (stdout and stack trace below) from bundler indicating that it couldn't reach https://rubygems,org/.

[stdout.txt](https://github.com/chef/supermarket/files/1604568/stdout.txt)
[stacktrace.txt](https://github.com/chef/supermarket/files/1604569/stacktrace.txt)

This system has rubygems_url set to our private gemserver in /etc/chef/client.rb, but supermarket seems to be ignoring that setting. I was able to get the reconfigure step to finish by changing the default gemserver on line 1055 of `/opt/supermarket/embedded/lib/ruby/gems/2.4.0/gems/chef-config-13.6.4/lib/chef-config/config.rb` from rubygems.org to our private gemserver - after that, the reconfigure ran just fine and the supermarket is back up and running now.

Am I missing something configuration-wise, or is this a regression since 3.1.42 (the last supermarket release I was successfully able to upgrade to)? I couldn't find a way to specify rubygems_url for supermarket specifically.

**Comment by robbkidd** [comment]
Quick update: figured out the culprit. Will follow-up in the next day with details.

**Comment by robbkidd** [comment]
*TL;DR:* I'm working on getting the cookbook internal to the Supermarket omnibus package to not use cookbooks that require gems.

### How did this come to pass?

The omnibus-internal cookbook `omnibus-supermarket` that drives the action of a `supermarket-ctl reconfigure` has for a long time required the `omnibus` cookbook. `omnibus-supermarket` was being used for two purposes: the reconfigure activity and a couple recipes to perform the actual omnibus build itself. In late Nov 2017, [the omnibus cookbook moved its requirement of the `pkg-config` gem](https://github.com/chef-cookbooks/omnibus/commit/8a19eb4efb0a11753d5d73949f4962c00bc91dfb) from a converge-time `chef_gem` resource, to a very-early-in-the-chef-run-all-the-time `gem` metadata entry. Builds of Supermarket after 3.1.46 have this newer `omnibus` cookbook which triggers a bundle install because of the metadata gem requirement.

### Good news!

We totally don't need to have the `omnibus-supermarket` cookbook serve dual purposes. I'll create a new cookbook for building and remove from `omnibus-supermarket` the dependency on `omnibus` cookbook which will remove the only cookbook with a gem dependency used by `omnibus-supermarket` for `reconfigure`.

**Comment by flakzilla** [comment]
That's some good sleuthing right there, thanks for helping us poor air-gapped peasants. We'll get up onto this release ASAP once it comes out.

**Comment by robbkidd** [comment]
@flakzilla Fix released in [v3.1.56](https://downloads.chef.io/supermarket/stable/3.1.56)!

**Comment by flakzilla** [comment]
All is well for our Supermarket, thanks again Robb!
